fusequery/query/src/api/rpc/flight_dispatcher.rs

```diff
@@ -0,0 +1,369 @@
+// Copyright 2020-2021 The Datafuse Authors.
+//
+// SPDX-License-Identifier: Apache-2.0.
+
+use std::collections::HashMap;
+use std::convert::TryInto;
+use std::sync::Arc;
+
+use common_arrow::arrow::datatypes::SchemaRef;
+use common_arrow::arrow::record_batch::RecordBatch;
+use common_arrow::arrow_flight::FlightData;
+use common_datavalues::DataSchemaRef;
+use common_exception::ErrorCodes;
+use common_exception::Result;
+use common_planners::Expression;
+use common_planners::PlanNode;
+use log::error;
+use tokio::sync::mpsc::channel;
+use tokio::sync::mpsc::Receiver;
+use tokio::sync::mpsc::Sender;
+use tokio_stream::StreamExt;
+
+use crate::api::rpc::flight_scatter::FlightScatter;
+use crate::clusters::ClusterRef;
+use crate::configs::Config;
+use crate::pipelines::processors::Pipeline;
+use crate::pipelines::processors::PipelineBuilder;
+use crate::sessions::FuseQueryContextRef;
+use crate::sessions::SessionManagerRef;
+
+#[derive(Debug)]
+pub struct PrepareStageInfo(
```

Comment:
Prefer to use a nomal struct for better code reading instead the tuple struct.
In particular, the `prepare_stage` method some inconvenient to read